lib/fiddle/struct.rb

```diff
@@ -119,8 +140,18 @@ class CStructEntity < Fiddle::Pointer
     #
     # See Fiddle::Pointer.malloc for memory management issues.
     def CStructEntity.malloc(types, func = nil)
-      addr = Fiddle.malloc(CStructEntity.size(types))
-      CStructEntity.new(addr, types, func)
+      addr = Fiddle.malloc(self.size(types))
+      struct = new(addr, types, func)
```

Comment:
If an exception is raised in here, `addr` isn't freed.

Reply:
Yeah maybe I should use the pointer constructor rather than using `malloc` directly as it used to.

Reply:
Fixed.